src/graph.rs

```diff
@@ -974,6 +974,43 @@ impl<T> Graph<T> {
         Bfs::new(self)
     }
 
+    /// Returns an iterator over the vertices
+    /// of the graph which follows a DFS based
+    /// topological order (Kahn's algorithm).
+    ///
+    /// Topological sorting is not possible for
```

Comment:
```
Topological sorting is not possible for
graphs which contain a cycle. You may
use topo.is_cylic() == false to verify
that your graph is a DAG.
```

What happens if the user doesn't check this? Maybe we should add a panic?

Reply:
You are right, that's a problem.

I've pushed a fix and added a test to exercise the panic code. I can't figure out how to make the test check for PANIC_MSG, so I hard-coded the text into the test. That's a bit annoying.